docs/conf.py

```diff
@@ -0,0 +1,140 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# This file does only contain a selection of the most common options. For a
+# full list see the documentation:
+# http://www.sphinx-doc.org/en/master/config
+
+import datetime
+import os
+
+# The full version, including alpha/beta/rc tags
+from ebtelplusplus import __version__
+
+# -- Project information -----------------------------------------------------
+
+
+release = __version__
+is_development = '.dev' in __version__
+
+project = "ebtelplusplus"
+author = "Will Barnes"
+copyright = f"{datetime.datetime.now().year}, {author}"  # noqa: A001
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named "sphinx.ext.*") or your custom
+# ones.
+extensions = [
+    "sphinxcontrib.bibtex",
+    "sphinx.ext.autodoc",
+    "sphinx.ext.intersphinx",
+    "sphinx.ext.todo",
+    "sphinx.ext.coverage",
+    "sphinx.ext.inheritance_diagram",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.napoleon",
+    "sphinx.ext.doctest",
+    "sphinx.ext.mathjax",
+    "sphinx_automodapi.automodapi",
+    "sphinx_automodapi.smart_resolver",
+    "sphinx_gallery.gen_gallery",
+    "sphinx_design",
+    "sphinx_copybutton",
+]
+
+# Add any paths that contain templates here, relative to this directory.
+# templates_path = ["_templates"]
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
+
+# The suffix(es) of source filenames.
+# You can specify multiple suffix as a list of string:
+source_suffix = ".rst"
+
+# The master toctree document.
+master_doc = "index"
+
+# Treat everything in single ` as a Python reference.
+default_role = 'py:obj'
+
+# -- Options for intersphinx extension ---------------------------------------
```

Comment:
Do you want to enable nitpicky? Maybe not in this PR tho.

Reply:
Yeah I'll leave it to a subsequent PR.